src/widget/wvumeter.cpp

```diff
@@ -148,88 +174,153 @@ void WVuMeter::updateState(mixxx::Duration elapsed) {
     m_dPeakParameter = math_clamp(m_dPeakParameter, 0.0, 1.0);
 }
 
-void WVuMeter::maybeUpdate() {
-    if (m_dParameter != m_dLastParameter || m_dPeakParameter != m_dLastPeakParameter) {
-        repaint();
-    }
+void WVuMeter::paintEvent(QPaintEvent* e) {
+    Q_UNUSED(e);
 }
 
-void WVuMeter::paintEvent(QPaintEvent * /*unused*/) {
-    ScopedTimer t("WVuMeter::paintEvent");
+void WVuMeter::showEvent(QShowEvent* e) {
+    Q_UNUSED(e);
+    // find the base color recursively in parent widget
+    m_qBgColor = findBaseColor(this);
```

Comment:
Here I added
`m_bHasRendered = false;`
to get rid of artifacts after hiding and showing the mixer in LateNight.

Reply:
Adding that in `paintEvent()` only fixes the incomplete initial rendering.

Reply:
right! thanks!